cmd/kube-proxy/app/server.go

```diff
@@ -1086,22 +1098,21 @@ func detectNodeIPs(ctx context.Context, rawNodeIPs []net.IP, bindAddress string)
 
 // getNodeIP returns IPs for the node with the provided name.  If
 // required, it will wait for the node to be created.
-func getNodeIPs(ctx context.Context, client clientset.Interface, name string) []net.IP {
+func getNodeIPs(ctx context.Context, nodeLister corelisters.NodeLister, name string) []net.IP {
 	logger := klog.FromContext(ctx)
 	var nodeIPs []net.IP
-	backoff := wait.Backoff{
-		Steps:    6,
-		Duration: 1 * time.Second,
-		Factor:   2.0,
-		Jitter:   0.2,
-	}
 
-	err := wait.ExponentialBackoff(backoff, func() (bool, error) {
-		node, err := client.CoreV1().Nodes().Get(ctx, name, metav1.GetOptions{})
+	err := wait.PollUntilContextCancel(ctx, 1*time.Second, true, func(context.Context) (bool, error) {
+		node, err := nodeLister.Get(name)
 		if err != nil {
 			logger.Error(err, "Failed to retrieve node info")
 			return false, nil
 		}
+		// don't consider the node if is going to be deleted and keep waiting
+		if !node.DeletionTimestamp.IsZero() {
+			logger.Info("Node is being deleted", "node", node.Name)
+			return false, nil
+		}
 		nodeIPs, err = utilnode.GetNodeHostIPs(node)
 		if err != nil {
 			logger.Error(err, "Failed to retrieve node IPs")
```

Comment:
This should be `Info` (though with the `err` included). It most likely just indicates that we fetched the Node in between when kubelet created it and when the cloud provider filled in the IPs.